content/influxdb/cloud-dedicated/query-data/troubleshoot-and-optimize/analyze-query-plan.md

```diff
@@ -0,0 +1,786 @@
+  predicate=time@5 >= 200 AND time@5 < 700 AND state@4 = MA,
+  pruning_predicate=time_max@0 >= 200 AND time_min@1 < 700 AND state_min@2 <= MA AND MA <= state_max@3
+```
+
+{{% caption %}}ParquetExec_B, the second ParquetExec{{% /caption %}}
+
+Because ParquetExec_B has overlaps, the `projection` and `output_ordering` expressions use the `__chunk_order` column used in [`RecordBatchesExec` `projection`](#projection-1).
+
+{{% note %}}
+The presence of `__chunk_order` in data scanning nodes indicates that data overlaps, and is possibly duplicated, among the nodes.
+{{% /note %}}
+
+The remaining ParquetExec_B expressions are similar to those in [ParquetExec_A](#parquetexec_a).
+
+##### How a query plan distributes data for scanning
+
+If you compare [`file_group`](#file_groups) paths in [ParquetExec_A](#parquetexec_a) to those in [ParquetExec_B](#parquetexec_b), you'll notice that both contain files from the same partition:
+
+{{% code-callout "b862a7e9b329ee6a4..." %}}
+
+```text
+1/1/b862a7e9b329ee6a4.../...
+```
+
+{{% /code-callout %}}
+
+The planner may distribute files from the same partition to different scan nodes for several reasons, including optimizations for handling [overlaps](#how-a-query-plan-distributes-data-for-scanning)--for example:
+
+- to separate non-overlapped files from overlapped files to minimize work required for deduplication (which is the case in this example)
+- to distribute non-overlapped files to increase parallel execution
+
+### Analyze branch structures
+
+After data is output from a data scanning node, it flows up to the next parent (outer) node.
+
+In the example plan:
+
+- Each leaf node is the first step in a branch of nodes planned for processing the scanned data.
+- The three branches execute in parallel.
+- After the leaf node, each branch contains the following similar node structure:
+
+```sql
+...
+CoalesceBatchesExec: target_batch_size=8192
+    FilterExec: time@3 >= 200 AND time@3 < 700 AND state@2 = MA
+    ...
+```
+
+- `FilterExec: time@3 >= 200 AND time@3 < 700 AND state@2 = MA`: filters data for the condition `time@3 >= 200 AND time@3 < 700 AND state@2 = MA`, and guarantees that all data is pruned.
+- `CoalesceBatchesExec: target_batch_size=8192`: combines small batches into larger batches. See the DataFusion [`CoalesceBatchesExec`] documentation.
+
+#### Sorting yet-to-be-persisted data
+
+In the `RecordBatchesExec` branch, the node that follows `CoalesceBatchesExec` is a `SortExec` node:
+
+```sql
+SortExec: expr=[state@2 ASC,city@1 ASC,time@3 ASC,__chunk_order@0 ASC]
+```
+
+The node uses the specified expression `state ASC, city ASC, time ASC, __chunk_order ASC` to sort the yet-to-be-persisted data.
+Neither ParquetExec_A nor ParquetExec_B contain a similar node because data in the Object store is already sorted (by the Compactor) in the given order; the query plan only needs to sort data that arrives from the Ingester.
```

Comment:
```suggestion
Neither ParquetExec_A nor ParquetExec_B contain a `SortExec` node because data in the Object store is already sorted in the given order; the query plan only needs to sort data that arrives from the Ingester.
```